xgcm/test/test_exchange.py

```diff
@@ -167,12 +168,20 @@ def test_create_connected_grid(ds, ds_face_connections_x_to_x):
     assert xaxis._connections[1][0][1] is xaxis
 
 
-@pytest.mark.xfail(reason="connected grids not implemented for grid ufunc refactor yet")
+def test_create_connected_grid_missing_face(ds, ds_face_connections_x_to_x):
+    # remove one of the faces and make sure an exception is raised.
+    del ds_face_connections_x_to_x["face"][1]
+    with pytest.raises(KeyError):
+        Grid(ds, face_connections=ds_face_connections_x_to_x)
+
+
 def test_diff_interp_connected_grid_x_to_x(ds, ds_face_connections_x_to_x):
     # simplest scenario with one face connection
-    grid = Grid(ds, face_connections=ds_face_connections_x_to_x)
-    diff_x = grid.diff(ds.data_c, "X", boundary="fill")
-    interp_x = grid.interp(ds.data_c, "X", boundary="fill")
+    grid = Grid(ds, face_connections=ds_face_connections_x_to_x, periodic=False)
```

Comment:
@rabernat I am very confused here. The original [test](https://github.com/xgcm/xgcm/blob/2ecc59cbd0af9ac1d21eee9c2b18a639cf0e02b4/xgcm/test/test_exchange.py#L170-L186) did not specify `periodic=False`. Was the expected behavior that setting boundary in a grid method overwrites periodic?

Eventually we will not have this ambiguity anymore, when we move periodic as an argument to `boundary`/`padding`, but I wanted to make this PR as backwards compatible as possible.